### PR TITLE
feat: added automatic shell detection

### PR DESCRIPTION
**Pull request by heinwol**
so when you run `, -s <pname>` you'll be sent into the shell you were in, not into bash.

Idk what else to say, please ask any questions you want if the matter is unclear

EDIT: oh, I remembered: parent pids detection is a bit hardcoded, I'm not sure if it would work on mac (I never used them)

### CODE REVIEW

File: src/shell.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+use std::{error::Error, fs};
+
+type ResultDyn<T> = Result<T, Box<dyn Error>>;
+
+const KNOWN_SHELLS: &[&str] = &[
+    "bash",   //
```
Comment (SuperSandro2000) [inline src/shell.rs:6]:
```suggestion
    "ash",
    "bash",   //
```

Comment (heinwol) [reply src/shell.rs:7]: Done!

**Comment by heinwol** [comment]
So, what do you guys think?

**Comment by heinwol** [comment]
@Artturin, sorry for pinging, just a reminder

**Review by heinwol — COMMENTED** [review]
Oh, dumb mistakes indeed, thanks

**Pull request merged**